js/src/components/paid-ads/convertToAssetGroupUpdateBody.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Internal dependencies
+ */
+import { ASSET_KEY, ASSET_FORM_KEY } from '.~/constants';
+
+/**
+ * @typedef {import('.~/components/types.js').AssetGroupFormValues} AssetGroupFormValues
+ * @typedef {import('.~/data/types.js').AssetEntityGroup} AssetEntityGroup
+ * @typedef {import('.~/data/types.js').AssetOperations} AssetOperations
+ * @typedef {import('.~/data/types.js').AssetEntityGroupUpdateBody} AssetEntityGroupUpdateBody
+ */
+
+/**
+ * Differences the assets form values against the existing asset entity group and
+ * returns the updating operations of the asset group.
+ *
+ * @param  {AssetEntityGroup} assetGroup The existing asset entity group as the base data for comparing the differences.
+ * @param  {AssetGroupFormValues} values The assets form values to be compared.
+ * @return {AssetOperations[]} The creation and deletion operations for updating the asset group.
+ */
+export function diffAssetOperations( assetGroup, values ) {
+	const operations = [];
+	const assets = assetGroup.assets;
+
+	function pushDeletions( key, ...assetEntities ) {
+		const deletions = assetEntities.map( ( assetEntity ) => ( {
+			...assetEntity,
+			// Setting the `content` to null indicates the asset deletion operation.
+			content: null,
+			field_type: key,
+		} ) );
+		operations.push( ...deletions );
+	}
+
+	Object.values( ASSET_KEY ).forEach( ( key ) => {
+		const contents = [ values[ key ] ].flat().filter( Boolean );
```

Comment:
I also would recommend to change here the `values` and `contents` generic for something more meaningful.

Reply:
Could you explain more thoughts? Especially the `contents` are the array of `content` values which is the needed name for the API request body, and the primary purpose of this function is to generate the part of mutation operations in the request body. I don't pretty understand why it's considered generic.

Reply:
In general my troubles come when I read the function (maybe because I'm reading it in GITHUB file changes page)

I read first what content/values variables are holding. Then I follow reading and at some point my mind is like "Wait a moment, what `content` was, and what `values` had? Values was the form ones or the previous assets?" So I go up again and need to confirm what that variable was holding. 

Again, maybe is because I was reading that here in GH instead of my IDE and at night (slow brain) but still I think `content` and `values` are a bit meaningless in the diff logic context. In comparisons I usually like to use `previousValues` / `newValues` or something equivalent. But in previous discussions seems you disliked this approach.

Reply:
Sorry, I hardly follow the reason that if any variable names are considered confusing just because the naming is hard to read in GH's file change page.

> [...] but still I think `content` and `values` are a bit meaningless in the diff logic context. In comparisons I usually like to use `previousValues` / `newValues` or something equivalent. But in previous discussions seems you disliked this approach.

It has no bearing on whether I like the approach or not, but just its context is not what you mentioned.

There is only one instance of the form values in this function, so it doesn't have the relative concept or transient status like previous or new. From the perspective of this function, it only needs two different data sources and returns the diffs. Using the naming with a relative concept like `previousValues`, `newValues` would make it very confusing instead.